tests/naming/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![allow(dead_code)]
+#[warn(non_camel_case_types)]
```

Comment:
I was thinking about local attributes, but that would clutter the file, but indeed a module attribute is nicer here. But then I think you need to add a bang:  
```suggestion
#![warn(non_camel_case_types)]
```

(otherwise it's only applying to `Foo`)